### PR TITLE
build: GNU/kFreeBSD support

### PR DESCRIPTION
**Pull request by h0tw1r3**
autotools support only, gvp does not support kfreebsd detection.

Update PR to hopefully meet contributing guidelines.
Original PR: https://github.com/libuv/libuv/pull/959

### CODE REVIEW

**Comment by saghul** [comment]
LGTM I'll run a CI build when it works, it's apparently broken now (the CI server): https://github.com/nodejs/build/issues/457

**Comment by cjihrig** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/80/

**Comment by saghul** [comment]
@h0tw1r3 can you please rebase? I think this is ready to land. Maybe @bnoordhuis can take a look if he has the time.

**Comment by h0tw1r3** [comment]
@saghul ping

**Comment by saghul** [comment]
One more CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/101/

File: src/unix/core.c [hunk 1]
```diff
@@ -55,7 +55,7 @@
 # endif
 #endif
 
-#if defined(__FreeBSD__) || defined(__DragonFly__)
+#if defined(__FreeBSD__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
```
Comment (bnoordhuis) [inline src/unix/core.c]: Long line, can you wrap this at 80 columns?

File: src/unix/fs.c [hunk 2]
```diff
@@ -595,7 +597,10 @@ static ssize_t uv__fs_sendfile(uv_fs_t* req) {
 
     return -1;
   }
-#elif defined(__FreeBSD__) || defined(__APPLE__) || defined(__DragonFly__)
+#elif defined(__APPLE__)           || \
+      defined(__FreeBSD__)         || \
+      defined(__DragonFly__)       || \
+      defined(__FreeBSD_kernel__)
```
Comment (bnoordhuis) [inline src/unix/fs.c]: For consistency, can you put the `__FreeBSD__` and `__FreeBSD_kernel__` guards together?

File: src/unix/fs.c [hunk 3]
```diff
@@ -608,6 +613,9 @@ static ssize_t uv__fs_sendfile(uv_fs_t* req) {
 #if defined(__FreeBSD__) || defined(__DragonFly__)
     len = 0;
     r = sendfile(in_fd, out_fd, req->off, req->bufsml[0].len, NULL, &len, 0);
+#elif defined(__FreeBSD_kernel__)
+    len = 0;
+    r = bsd_sendfile(in_fd, out_fd, req->off, req->bufsml[0].len, NULL, &len, 0);
```
Comment (bnoordhuis) [inline src/unix/fs.c]: Long line.  Please line up the arguments when you break it up.

File: src/unix/process.c [hunk 4]
```diff
@@ -40,7 +40,7 @@
 extern char **environ;
 #endif
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__GLIBC__)
```
Comment (bnoordhuis) [inline src/unix/process.c]: I suppose this raises the interesting question if e.g. kfreebsd + musl works / should work.

File: test/test-tty.c [hunk 5]
```diff
@@ -260,8 +260,14 @@ TEST_IMPL(tty_file) {
...
-# if defined(__linux__) && !defined(__ANDROID__) || defined(__OpenBSD__) || defined(__NetBSD__) || \
-    defined(__APPLE__) || defined(__FreeBSD__) || defined(__DragonFly__)
+#if defined(__linux__)           || \
+    defined(__ANDROID__)         || \
+    defined(__OpenBSD__)         || \
+    defined(__NetBSD__)          || \
+    defined(__APPLE__)           || \
+    defined(__FreeBSD__)         || \
+    defined(__DragonFly__)       || \
+    defined(__FreeBSD_kernel__)
```
Comment (bnoordhuis) [inline test/test-tty.c]: Can you put the guards together?

**Comment by bnoordhuis** [comment]
LGTM minus nits.

**Comment by h0tw1r3** [comment]
@bnoordhuis @saghul addressed nits.

**Comment by saghul** [comment]
Landed in 0a4b51f, cheers! I see this is your first code contribution to libuv, welcome aboard :-)

**Pull request closed without merging**